Tabnine/src/main/java/com/tabnine/chat/ChatEnabledState.kt

```diff
@@ -0,0 +1,72 @@
+package com.tabnine.chat
+
+import com.intellij.openapi.Disposable
+import com.intellij.util.messages.Topic
+import com.tabnineCommon.capabilities.CapabilitiesService
+import com.tabnineCommon.capabilities.Capability
+import com.tabnineCommon.chat.ChatDisabledReason
+import com.tabnineCommon.chat.ChatFrame
+import com.tabnineCommon.chat.ChatState
+import com.tabnineCommon.general.TopicBasedNonNullState
+import com.tabnineCommon.lifecycle.BinaryStateSingleton
+import com.tabnineCommon.lifecycle.CapabilitiesStateSingleton
+import java.util.function.Consumer
+
+class ChatEnabledState private constructor() : ChatFrame.UseChatEnabledState,
+    TopicBasedNonNullState<ChatState, ChatEnabledChanged>(
+        ENABLED_TOPIC, ChatState.loading()
+    ) {
+
+    companion object {
+        private val ENABLED_TOPIC: Topic<ChatEnabledChanged> =
+            Topic.create("ChatEnabled", ChatEnabledChanged::class.java)
+
+        val instance = ChatEnabledState()
+    }
+
+    init {
+        updateEnabled()
+
+        BinaryStateSingleton.instance.useState {
+            updateEnabled()
+        }
+
+        CapabilitiesStateSingleton.instance.useState {
+            updateEnabled()
+        }
+    }
+
+    private fun updateEnabled() {
+        if (!CapabilitiesService.getInstance().isReady) {
+            return
+        }
+
+        val isLoggedIn = BinaryStateSingleton.instance.get()?.isLoggedIn ?: return
+
+        val alphaEnabled = CapabilitiesService.getInstance().isCapabilityEnabled(Capability.ALPHA)
+        val chatCapabilityEnabled =
+            CapabilitiesService.getInstance().isCapabilityEnabled(Capability.TABNINE_CHAT)
```

Comment:
how does this behave in SH? empty responses?

Reply:
It doesn't execute this code in SH, this code is only for the SAAS plugin. Anyhow the Capabilities service returns true for any capability asked in SH (I always was that way, didn't change it).